app/src/main/java/com/jerboa/ui/components/common/InputFields.kt

```diff
@@ -584,7 +584,8 @@ fun MarkdownHelperBar(
         IconButton(
             onClick = {
                 simpleMarkdownSurround(
-                    "~",
+                    startText = "<sub>",
```

Comment:
This isn't the right way to do it, we have to follow the conventions [here](https://join-lemmy.org/docs/users/02-media.html?highlight=markdown#text) (but it looks like I need to fix super and subscript there also) , so that all lemmy clients can support it. 

This needs to be fixed in the markdown renderer.

The best way to test it to use lemmy-ui, and make sure the rendered, and view source, match both on jerboa and lemmy-ui.

Reply:
I'm fairly sure also that these already exist in [eternity](https://codeberg.org/Bazsalanszky/Eternity), and could be taken from there.

Reply:
I checked eternity only supports superscript and its slightly different syntax too. its just `^`

![image](https://github.com/dessalines/jerboa/assets/67873169/746f19a0-6731-4dc9-bb1b-0bb0ddff6055)

Their execution is very complex. 

Imo I think we can achieve something better with a plugin that just rewrites ^^ and ~~ to their html tags equivalent. But from user perspective it will look like its `^^`. That way we wont need to maintain all this super complex logic that seems to depend on other custom markwon nodes.

Reply:
Totally agree. There's a simple example of altering the markdown string before processing here: https://noties.io/Markwon/docs/v4/core/plugins.html#process-markdown

It'd be much easier to do regex to replace `^...^` with `<sup>...</sup>`.